cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/to/Nexus2SearchResponse.java

```diff
@@ -0,0 +1,175 @@
+package com.devonfw.cobigen.api.util.to;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.devonfw.cobigen.api.constants.MavenSearchRepositoryConstants;
+import com.devonfw.cobigen.api.constants.MavenSearchRepositoryType;
+import com.devonfw.cobigen.api.exception.RESTSearchResponseException;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Json model for nexus2 Search REST API response
+ *
+ */
+@JsonIgnoreProperties(value = { "totalCount", "from", "count", "tooManyResults", "collapsed", "repoDetails" })
+public class Nexus2SearchResponse implements SearchResponse {
+
+  /** Logger instance. */
+  @JsonIgnore
+  private static final Logger LOG = LoggerFactory.getLogger(Nexus2SearchResponse.class);
+
+  @JsonProperty("data")
+  private List<Nexus2SearchResponseData> data;
+
+  @Override
+  @JsonIgnore
+  public List<URL> getDownloadURLs() throws MalformedURLException {
+
+    List<URL> downloadLinks = new ArrayList<>();
+
+    for (Nexus2SearchResponseData item : this.data) {
+      for (Nexus2SearchResponseArtifactHits artifactHit : item.artifactHits) {
+        for (Nexus2SearchResponeArtifactLinks artifactLink : artifactHit.artifactLinks) {
+          downloadLinks.add(SearchResponseUtil.createDownloadLink(
```

Comment:
sounds like the methods of `SearchResponseUtil` can and should be easily integrated into `SearchResponse` class

Reply:
Moved all methods to AbstractSearchResponse now.